packages/flutter/lib/src/painting/box_border.dart

```diff
@@ -489,6 +503,22 @@ class Border extends BoxBorder {
         && right.strokeAlign == topStrokeAlign;
   }
 
+  Set<Color> get _distinctVisibleColors => <Color?>{
+        if (top.style == BorderStyle.none) null else top.color,
+        if (right.style == BorderStyle.none) null else right.color,
+        if (bottom.style == BorderStyle.none) null else bottom.color,
+        if (left.style == BorderStyle.none) null else left.color,
+      }.whereNotNull().toSet();
```

Comment:
this would probably be more efficient as:

```suggestion
  Set<Color> get _distinctVisibleColors => <Color>{};
  if (top.style != BorderStyle.none) _distinctVisibleColors.add(top.color);
  if (right.style != BorderStyle.none) _distinctVisibleColors.add(right.color);
  if (bottom.style != BorderStyle.none) _distinctVisibleColors.add(bottom.color);
  if (left.style != BorderStyle.none) _distinctVisibleColors.add(left.color);
```

Reply:
I guess it can't be a `final` field. Here's an alternative that is closer to your original but still avoids the nulls and iterators:

```suggestion
  Set<Color> get _distinctVisibleColors => <Color>{
        if (top.style != BorderStyle.none) top.color,
        if (right.style != BorderStyle.none) right.color,
        if (bottom.style != BorderStyle.none) bottom.color,
        if (left.style != BorderStyle.none) left.color,
      };
```

Reply:
It can't be a getter :( so it would need to be a function. But it is fine, I can make it a function. I guess I was optimizing for readability, but this works too.

Reply:
not sure what you mean, the last one suggested here a getter?

Reply:
I liked getters lol. I mean, there were 50 getters, I tried to make another one, but function is fine.